include/deal.II/base/template_constraints.h

```diff
@@ -466,6 +466,40 @@ struct EnableIfScalar<std::complex<T>>
 };
 
 
+
+template <typename U, typename... T>
+constexpr bool contains(std::tuple<T...>)
+{
+  return !std::is_same<
+    std::integer_sequence<bool, false, std::is_same<U, T>::value...>,
+    std::integer_sequence<bool, std::is_same<U, T>::value..., false>>::value;
+}
+
+
+
+template <typename T>
+constexpr bool
+is_mpi_type()
+{
+  return contains<T>(std::tuple<bool,
+                                char,
+                                signed char,
+                                short,
+                                int,
+                                long int,
+                                unsigned char,
+                                unsigned short,
+                                unsigned int,
+                                unsigned long int,
+                                unsigned long long int,
+                                float,
+                                double,
+                                long double,
+                                std::complex<float>,
+                                std::complex<double>>{});
+}
```

Comment:
I think this should rather be in `mpi.h`, possibly also in `internal` so that we don't promise its existence. :slightly_smiling_face:

Reply:
Fixed in b36cdf4